### ISSUE TITLE
Raw ADC Radar data UWCR dataset several questions about the data to train on.

### ISSUE THREAD

**Issue by jimvermunt** [Xiangyu-Gao/Radar-multiple-perspective-object-detection#16]
In the paper "RAMP-CNN: A Novel Neural Network for Enhanced Automotive Radar Object Recognition" it is mentioned that the training and test set have 67,198 and 25,098 training and testing sets, totaling 92,296 frames. 

I checked with the [raw radar set](https://github.com/Xiangyu-Gao/Raw_ADC_radar_dataset_for_automotive_object_detection) and only got 23 sequences in total, which each consist of 899 frames, totaling 20,677 frames. This is only 1/5th of the data that is mentioned in the paper. Is the raw radar set a subset of the data where the RAMP-CNN model is trained? 

Could you elaborate more on what the differences are?

**Comment by jimvermunt** [comment]
And what are the sequences used in the following set for training and testing?
~~~
sequences_all = ['2019_05_28_pm2s012','2019_05_29_bcms000', '2019_04_09_css1000', '2019_04_30_cm1s000', '2019_05_29_pcms005', '2019_05_09_mlms003', '2019_04_30_mlms000', '2019_05_09_cm1s003', '2019_04_09_pms1000', '2019_04_09_pms3000', '2019_05_09_pbms004', '2019_05_29_mlms006', '2019_05_09_bm1s007', '2019_04_09_pms2000', '2019_04_09_bms1000', '2019_05_09_pcms002', '2019_05_29_pbms007', '2019_04_30_mlms001', '2019_04_30_pbms002', '2019_04_09_cms1000', '2019_04_30_pcms001', '2019_04_30_pbss000', '2019_05_29_cm1s014']
~~~

**Comment by jimvermunt** [comment]
From the [raw radar set UWCR](https://github.com/Xiangyu-Gao/Raw_ADC_radar_dataset_for_automotive_object_detection) in the sequence 2019_04_09_pms2000 it has 0-897 images, 2-899 raw frames, and 3-897 text labels. How does it come that these are misaligned?

This also holds for other sequences but they have different sizes of misalignment.



~~~
Dir sequence ../data/Automotive/2019_05_29_bcms000
Number of images: 900
Number of radar frames: 898
Number of text labels: 897
Diff images and radar frames: 2
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000899.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000899.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: 0
Diff last image and last text label: 0
1
Dir sequence ../data/Automotive/2019_04_09_css1000
Number of images: 897
Number of radar frames: 897
Number of text labels: 894
Diff images and radar frames: 0
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000896.jpg
Id of first and last radar frame: 000003.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000896.csv
Diff first image and first radar frame: -3
Diff first image and first text label: -3
Diff last image and last radar frame: -3
Diff last image and last text label: 0
2
Dir sequence ../data/Automotive/2019_04_30_cm1s000
Number of images: 900
Number of radar frames: 898
Number of text labels: 896
Diff images and radar frames: 2
Diff images and text labels: 4
Id of first and last image: 0000000000.jpg 0000000899.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000898.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: 0
Diff last image and last text label: 1
3
Dir sequence ../data/Automotive/2019_05_29_pcms005
Number of images: 900
Number of radar frames: 898
Number of text labels: 897
Diff images and radar frames: 2
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000899.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000899.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: 0
Diff last image and last text label: 0
4
Dir sequence ../data/Automotive/2019_05_09_mlms003
Number of images: 900
Number of radar frames: 898
Number of text labels: 897
Diff images and radar frames: 2
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000899.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000899.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: 0
Diff last image and last text label: 0
5
Dir sequence ../data/Automotive/2019_04_30_mlms000
Number of images: 900
Number of radar frames: 898
Number of text labels: 897
Diff images and radar frames: 2
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000899.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000899.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: 0
Diff last image and last text label: 0
6
Dir sequence ../data/Automotive/2019_05_09_cm1s003
Number of images: 900
Number of radar frames: 898
Number of text labels: 897
Diff images and radar frames: 2
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000899.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000899.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: 0
Diff last image and last text label: 0
7
Dir sequence ../data/Automotive/2019_04_09_pms1000
Number of images: 898
Number of radar frames: 898
Number of text labels: 895
Diff images and radar frames: 0
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000897.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000897.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: -2
Diff last image and last text label: 0
8
Dir sequence ../data/Automotive/2019_04_09_pms3000
Number of images: 898
Number of radar frames: 898
Number of text labels: 895
Diff images and radar frames: 0
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000897.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000897.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: -2
Diff last image and last text label: 0
9
Dir sequence ../data/Automotive/2019_05_09_pbms004
Number of images: 900
Number of radar frames: 898
Number of text labels: 897
Diff images and radar frames: 2
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000899.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000899.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: 0
Diff last image and last text label: 0
10
Dir sequence ../data/Automotive/2019_05_29_mlms006
Number of images: 900
Number of radar frames: 898
Number of text labels: 897
Diff images and radar frames: 2
Diff images and text labels: 3
Id of first and last image: 0000000000.jpg 0000000899.jpg
Id of first and last radar frame: 000002.mat 000899.mat
Id of first and last text label: 0000000003.csv 0000000899.csv
Diff first image and first radar frame: -2
Diff first image and first text label: -3
Diff last image and last radar frame: 0
Diff last image and last text label: 0
~~~

**Comment by Xiangyu-Gao** [comment]
> In the paper "RAMP-CNN: A Novel Neural Network for Enhanced Automotive Radar Object Recognition" it is mentioned that the training and test set have 67,198 and 25,098 training and testing sets, totaling 92,296 frames.
> 
> I checked with the [raw radar set](https://github.com/Xiangyu-Gao/Raw_ADC_radar_dataset_for_automotive_object_detection) and only got 23 sequences in total, which each consist of 899 frames, totaling 20,677 frames. This is only 1/5th of the data that is mentioned in the paper. Is the raw radar set a subset of the data where the RAMP-CNN model is trained?
> 
> Could you elaborate more on what the differences are?

Yes, this raw radar set is a subset of the data where the RAMP-CNN model is trained & tested.

**Comment by Xiangyu-Gao** [comment]
> And what are the sequences used in the following set for training and testing?
> 
> ```
> sequences_all = ['2019_05_28_pm2s012','2019_05_29_bcms000', '2019_04_09_css1000', '2019_04_30_cm1s000', '2019_05_29_pcms005', '2019_05_09_mlms003', '2019_04_30_mlms000', '2019_05_09_cm1s003', '2019_04_09_pms1000', '2019_04_09_pms3000', '2019_05_09_pbms004', '2019_05_29_mlms006', '2019_05_09_bm1s007', '2019_04_09_pms2000', '2019_04_09_bms1000', '2019_05_09_pcms002', '2019_05_29_pbms007', '2019_04_30_mlms001', '2019_04_30_pbms002', '2019_04_09_cms1000', '2019_04_30_pcms001', '2019_04_30_pbss000', '2019_05_29_cm1s014']
> ```



> 

'2019_05_28_pm2s012' is in the testing set for ramp-cnn paper, and others are in the training set.

**Comment by Xiangyu-Gao** [comment]
> From the [raw radar set UWCR](https://github.com/Xiangyu-Gao/Raw_ADC_radar_dataset_for_automotive_object_detection) in the sequence 2019_04_09_pms2000 it has 0-897 images, 2-899 raw frames, and 3-897 text labels. How does it come that these are misaligned?
> 
> This also holds for other sequences but they have different sizes of misalignment.
> 
> ```
> Dir sequence ../data/Automotive/2019_05_29_bcms000
> Number of images: 900
> Number of radar frames: 898
> Number of text labels: 897
> Diff images and radar frames: 2
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000899.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000899.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: 0
> Diff last image and last text label: 0
> 1
> Dir sequence ../data/Automotive/2019_04_09_css1000
> Number of images: 897
> Number of radar frames: 897
> Number of text labels: 894
> Diff images and radar frames: 0
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000896.jpg
> Id of first and last radar frame: 000003.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000896.csv
> Diff first image and first radar frame: -3
> Diff first image and first text label: -3
> Diff last image and last radar frame: -3
> Diff last image and last text label: 0
> 2
> Dir sequence ../data/Automotive/2019_04_30_cm1s000
> Number of images: 900
> Number of radar frames: 898
> Number of text labels: 896
> Diff images and radar frames: 2
> Diff images and text labels: 4
> Id of first and last image: 0000000000.jpg 0000000899.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000898.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: 0
> Diff last image and last text label: 1
> 3
> Dir sequence ../data/Automotive/2019_05_29_pcms005
> Number of images: 900
> Number of radar frames: 898
> Number of text labels: 897
> Diff images and radar frames: 2
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000899.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000899.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: 0
> Diff last image and last text label: 0
> 4
> Dir sequence ../data/Automotive/2019_05_09_mlms003
> Number of images: 900
> Number of radar frames: 898
> Number of text labels: 897
> Diff images and radar frames: 2
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000899.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000899.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: 0
> Diff last image and last text label: 0
> 5
> Dir sequence ../data/Automotive/2019_04_30_mlms000
> Number of images: 900
> Number of radar frames: 898
> Number of text labels: 897
> Diff images and radar frames: 2
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000899.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000899.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: 0
> Diff last image and last text label: 0
> 6
> Dir sequence ../data/Automotive/2019_05_09_cm1s003
> Number of images: 900
> Number of radar frames: 898
> Number of text labels: 897
> Diff images and radar frames: 2
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000899.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000899.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: 0
> Diff last image and last text label: 0
> 7
> Dir sequence ../data/Automotive/2019_04_09_pms1000
> Number of images: 898
> Number of radar frames: 898
> Number of text labels: 895
> Diff images and radar frames: 0
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000897.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000897.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: -2
> Diff last image and last text label: 0
> 8
> Dir sequence ../data/Automotive/2019_04_09_pms3000
> Number of images: 898
> Number of radar frames: 898
> Number of text labels: 895
> Diff images and radar frames: 0
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000897.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000897.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: -2
> Diff last image and last text label: 0
> 9
> Dir sequence ../data/Automotive/2019_05_09_pbms004
> Number of images: 900
> Number of radar frames: 898
> Number of text labels: 897
> Diff images and radar frames: 2
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000899.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000899.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: 0
> Diff last image and last text label: 0
> 10
> Dir sequence ../data/Automotive/2019_05_29_mlms006
> Number of images: 900
> Number of radar frames: 898
> Number of text labels: 897
> Diff images and radar frames: 2
> Diff images and text labels: 3
> Id of first and last image: 0000000000.jpg 0000000899.jpg
> Id of first and last radar frame: 000002.mat 000899.mat
> Id of first and last text label: 0000000003.csv 0000000899.csv
> Diff first image and first radar frame: -2
> Diff first image and first text label: -3
> Diff last image and last radar frame: 0
> Diff last image and last text label: 0
> ```
Use the filename as the timestamp to match the radar data, image, and labels. You can disregard the redundant ones.
